lib/features/academic_calendar/utils/localize_counter_message.dart

```diff
@@ -0,0 +1,17 @@
+import "package:flutter/cupertino.dart";
+
+import "../../../utils/context_extensions.dart";
+import "../model/academic_calendar_extensions.dart";
+import "../repository/academic_calendar_repo.dart";
+
+extension LocalizeCounterMessageExtension on AcademicCalendar {
+  String localizeDaysCounterMessage(BuildContext context) {
+    if (this.AcademicCalendarData!.isHolidays()) {
```

Comment:
I don't like the non null assertion `!` here :) maybe better to be safe than sorry?